src/cursor/abstract_cursor.ts

```diff
@@ -841,7 +841,10 @@ function cleanupCursor(
 
     if (session) {
       if (session.owner === cursor) {
-        return session.endSession({ error }, callback);
+        session.endSession({ error }).finally(() => {
+          callback();
+        });
+        return;
```

Comment:
Note from chat these returns getting moved down were to satisfy the `void` return type in TS.